backend/src/socket/socket.service.ts

```diff
@@ -0,0 +1,157 @@
+import { Server, Socket } from 'socket.io';
+import { Injectable } from '@nestjs/common';
+import { AuthService } from '../auth/auth.service';
+import { UserService } from '../users/user/user.service';
+import {
+	ChatId,
+	ChatList,
+	Chat_List_Item,
+	MessageList,
+	SingleMessage,
+	SocketMessage,
+} from './socket.types';
+
+@Injectable()
+export class SocketService {
+	constructor(
+		private readonly authService: AuthService,
+		private readonly userService: UserService,
+	) {}
+	public chatServer: Server = null;
+
+	public chatListSubscribers: ChatList = {};
+	public messageLists: MessageList = {};
+
+	chatIdToRoomName(chatId: string): string;
+	chatIdToRoomName(chatId: ChatId): string;
+	chatIdToRoomName(chatId: ChatId | string): string {
+		return `room_${chatId}`;
+	}
+
+	async joinRoom(chatId: ChatId, socket: Socket) {
+		const cookie: string = this.jwtCookieFromHandshakeString(
+			socket.handshake.headers.cookie,
+		);
+		try {
+			const userId: number = this.authService.userIdFromCookieString(cookie);
+			await this.userService.findOne({ where: { id: userId } });
+			const roomName = this.chatIdToRoomName(chatId);
+			if (!this.messageLists[chatId]) {
+				this.messageLists[chatId] = {
+					chatId: chatId,
+					roomName: roomName,
+					subscribers: [],
```

Comment:
does this one need a type?

Reply:
subscribers is typed,
in socket.types.ts you can find the definition for MessageList type and ChatInfo interface
```ts
export type UserId = number;

export interface ChatInfo {
	chatId: ChatId;
	roomName: string;
	subscribers: UserId[];
}

export type MessageList = Record<ChatId, ChatInfo>;
```